django/db/backends/postgresql/base.py

```diff
@@ -258,12 +258,11 @@ def _nodb_connection(self):
                 "and will use the default database instead.",
                 RuntimeWarning
             )
-            settings_dict = self.settings_dict.copy()
-            settings_dict['NAME'] = settings.DATABASES[DEFAULT_DB_ALIAS]['NAME']
             nodb_connection = self.__class__(
-                self.settings_dict.copy(),
+                {**self.settings_dict, 'NAME': settings.DATABASES[DEFAULT_DB_ALIAS]['NAME']},
                 alias=self.alias,
-                allow_thread_sharing=False)
+                allow_thread_sharing=False,
```

Comment:
Fine, but not as part of this PR. If you can change these cases systemically, that would be great.

Reply:
I wrote simple script and fixed hanging indentation in various code  #9790.